src/test/steps/loginSteps.ts

```diff
@@ -1,37 +1,42 @@
 import {Given, When, Then} from '@cucumber/cucumber';
-import {chromium, Page, Browser, expect} from '@playwright/test';
+import {pageFixture} from '../../hooks/pageFixture';
+import {expect} from '@playwright/test';
+import basePage from '../../pages/basePage';
 
-let browser: Browser;
-let page: Page;
+Given('User passes the authorization', async function () {
+  await basePage.authPage.authorizeUser();
 
-Given('User navigates to the application', async function () {
-  browser = await chromium.launch({headless: false});
-  page = await browser.newPage();
-  page.goto('https://px-stage.self-service.danads.com');
+  const pageTitle = await pageFixture.page.title();
+  expect(pageTitle).toContain('Core Ad Manager');
 });
 
-Given('User clicks login button', async function () {
-  //await page.locator("//a/*[text()='Log in']").click();
+Given('User clicks Log in button', async function () {
+  await basePage.loginPage.logInBtn.waitFor({state: 'visible', timeout: 30000});
+  await basePage.loginPage.logInBtn.click();
 });
 
-Given('User enters username', async function () {
-  await page.locator('(//input[@name=\'username\'])[2]').fill('Username');
+When('User enters email', async function () {
+  await basePage.loginPage.emailInput.waitFor({state: 'visible', timeout: 30000});
+  await basePage.loginPage.emailInput.fill(process.env.ADMIN_EMAIL);
```

Comment:
What if you need to log in as `Advertiser` or `Agency` in the test?

Reply:
Fixed